src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

```diff
@@ -0,0 +1,292 @@
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Helper class to place and remove constructionTapes from the buildings.
+ */
+public final class ConstructionTapeHelper
+{
+    public static final  PropertyDirection FACING     = BlockHorizontal.FACING;
+    public static final int MINHEIGHT = 1;
+    public static final int MAXHEIGHT = 256;
+    /**
+     * Private Constructor to hide implicit one.
+     */
+    private ConstructionTapeHelper()
+    {
+        /**
+         * Intentionally left empty.
+         */
+    }
+
+    /**
+     * Check if a block is placeable and return new Y position.
+     * @param x Block X position.
+     * @param y Block Y position.
+     * @param z Block Z position.
+     * @param world the world.
+     * @return The new Y position.
+     */
+
+    public static int checkIfPlaceable(@NotNull int x, @NotNull int y, @NotNull int z, @NotNull World world)
+    {
+        int newY = y;
+        boolean working = true;
+        while (working)
+        {
+            final BlockPos block = new BlockPos(x, newY, z);
+            final BlockPos blockMin1 = new BlockPos(x, newY-1, z);
+            if (world.getBlockState(block).getMaterial().isReplaceable())
+            {
+                if (world.getBlockState(blockMin1).getMaterial().isReplaceable())
+                {
+                    newY = newY-1;
+                }
+                else
+                {
+                    working = false;
+                }
+            }
+            else
+            {
+                newY = newY+1;
+            }
+        }
+        return newY;
+    }
+
+    /**
+     * @param world the world.
+     * @param block the block.
+     * @param tapeOrTapeCorner Is the checked block supposed to be ConstructionTape or ConstructionTapeCorner.
+     */
+    public static void removeTapeIfNecessary(@NotNull World world,@NotNull BlockPos block,@NotNull Block tapeOrTapeCorner)
+    {
+        for (int y = MINHEIGHT; y <= MAXHEIGHT; y++)
+        {
+            final BlockPos newBlock = new BlockPos(block.getX(), y, block.getZ());
+            if (world.getBlockState(newBlock).getBlock() == tapeOrTapeCorner)
+            {
+                world.setBlockState(newBlock, Blocks.AIR.getDefaultState());
+                break;
+            }
+        }
+    }
+
+    /**
+     * Proxy to place the tape also with the building only.
+     * @param building the building.
+     * @param world the world.
+     */
+    public static void placeConstructionTape(@NotNull AbstractBuilding building, @NotNull World world)
+    {
+            placeConstructionTape(new WorkOrderBuild(building, 1), world);
+    }
+
+    /**
+     * Place construction tape.
+     * @param workOrder the workorder.
+     * @param world the world.
+     */
+
+    public static void placeConstructionTape(@NotNull WorkOrderBuild workOrder, @NotNull World world)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "placeConstructionTape" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)